File: python/media3.py
notas = [float(x) for x in input().split()]
notas[0] *= 2
notas[1] *= 3
notas[2] *= 4
notas[3] *= 1

media = (notas[0] + notas[1] + notas[2] + notas[3]) / 10

if media >= 7:
    print('Media: {:.1f}'.format(media))
    print('Aluno aprovado.')
elif media < 5:
    print('Media: {:.1f}'.format(media))
    print('Aluno reprovado.')
else:
    aux = float(input())
    print('Media: {:.1f}'.format(media))
    print('Aluno em exame.')
    print('Nota do exame: {:.1f}'.format(aux))
    media = (media + aux) / 2
    if media >= 5:
        print('Aluno aprovado.')
    else:
        print('Aluno reprovado.')
    print('Media final: {:.1f}'.format(media))

File: java/media3.java
import java.util.Scanner;

public class media3 {
    public static void main(String[] args) {
        float notas[] = { 2, 3, 4, 1 };
        float media = 0;
        Scanner sc = new Scanner(System.in);
        for (int i = 0; i < 4; i++) {
            float aux;
            aux = sc.nextFloat();
            notas[i] *= aux;
            media += notas[i];
        }

        media /= 10;

        String str;
        if (media >= 7) {
            str = String.format("%.1f", media);
            System.out.println("Media: " + str);
            System.out.println("Aluno aprovado.");
        } else if (media < 5) {
            str = String.format("%.1f", media);
            System.out.println("Media: " + str);
            System.out.println("Aluno aprovado.");
        } else {
            float aux = sc.nextFloat();
            str = String.format("%.1f", media);
            System.out.println("Media: " + str);
            System.out.println("Aluno em exame.");
            str = String.format("%.1f", aux);
            System.out.println("Nota do exame: " + str);
            media = (media + aux) / 2;
            if (media >= 5) {
                System.out.println("Aluno aprovado.");
            } else {
                System.out.println("Aluno reprovado");
            }
            str = String.format("%.1f", media);
            System.out.println("Media final: " + str);
        }

        sc.close();
    }
}